## Scaling and generating gcode from `Toadstool Logo Scaleable 2.svg`

This is what Inkscape shows:

![image.png](./7f83b16.png)

so that's 657.39001 mm wide which is 25.8815 inches, which is pretty close to 24 already.

In [3]:
originalWidth = 657.39011 / 25.4
originalWidth

25.88150039370079

In [5]:
24 / originalWidth * 657.39011

609.6

so say 610 for grins.

Question is now how to scale it?

Inkscape has a "Scale" option on Document Properties -> Page, so computing the scaling factor is as follows:

In [6]:
610 /657.39011

0.9279117387391178

So this scale, when converted to a percentage may be put into Inkscape's Transform -> Scale tab as shown below:

![thing](./054999d.png)

I save this file as `Toadstool Logo Scaled to 24" wide.svg`

Now I need to figure out how to generate g-code for it using gcodetools.

According to Extensions -> Gcodetools -> About, the source for gcodetools may be found at http://www.cnc-club.ru/gcodetools

The available menu items in the version I have (1.7) are shown below:

![gcode tools menu](./2767639.png)

I've no idea where to start, so I've picked "Path to Gcode" and the following values on each of its three tabs ("Path to Gcode", "Options", "Preferences").  The fourth tab "Help" has no values.

![gcodetools Path to Gcode -> Path to Gcode](./2bf08ef.png)

![gcodetools Path to Gcode -> Options](./b26dc3d.png)

![gcodetools Path to Gcode -> Preferences](./681bc32.png)

When I hit Apply I got:

![Inkscape Error](./Screenshot_20240428_205710.png)

I added a layer (Layer 1) with nothing in it and hit Apply again:

![Inkscape Error](./Screenshot_20240428_205934.png)

However a file did appear:

output_001.ngc




In [1]:
cat output_0001.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z5.000000
G00 X586.714272 Y21.868317

G01 Z-0.125000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-0.125000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-0.125000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-0.125000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-0.125000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-0.125000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-0.125000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-0.125000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-0.125000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-0.125000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-0.125000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-0.125000 I8.004507 J19.089994
G02 X556.956

This line didn't seem quite right:

```
G01 Z-0.125000 F100.0(Penetrate)
```

That's 0.125 mm?

We need more like ${1 \over 4}$" which is 6.35 mm.

Not sure how to make that happen.  Maybe try Options -> "Scale along Z axis:" assuming that it's derived from "Minimum arc radius"?  It could have been generated from a default tool too I suppose.

Generated another .ngc file by hitting "Apply".

In [3]:
cat output_0002.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z255.000000
G00 X586.714272 Y21.868317

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-6.375000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-6.375000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-6.375000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-6.375000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-6.375000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-6.375000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-6.375000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-6.375000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-6.375000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-6.375000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-6.375000 I8.004507 J19.089994
G02 X556.9

This seemed to do the trick as far as the depth is concerned:

```
G01 Z-6.375000 F100.0(Penetrate)
```

Not sure if the Genmitsu 4030 can cut 6.375mm depth at 400mm/min though for the whole path.

Also, I still don't know if the depth value was derived from the "Minimum arc radius" or not.

To test this, I'll set "Minimum arc radius" to be 1mm instead of 0.125mm.

In [4]:
cat output_0003.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z255.000000
G00 X586.714272 Y21.868317

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-6.375000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-6.375000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-6.375000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-6.375000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-6.375000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-6.375000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-6.375000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-6.375000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-6.375000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-6.375000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-6.375000 I8.004507 J19.089994
G02 X556.9

No change, so it must be choosing a default tool for me, or getting the 0.125 default depth from somewhere else.

And this leads me to realize that for a ${1 \over 8}" diameter tool, the "Minimum arc radius:" must be 0.0625mm, so I set that as the default instead of 0.125mm.